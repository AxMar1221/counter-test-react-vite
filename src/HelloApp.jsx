// function App(){
//     return ( <h1>Hi Tachi</h1>);
// }
// export default App;

export const Helloapp = () => {
  return (
    <h1>Helloapp</h1>
  )
}
