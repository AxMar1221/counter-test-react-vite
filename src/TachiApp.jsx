import PropTypes from 'prop-types'

// const getSuma = (a, b) => {
//   return a + b;
// }

// const newValue = {
//   message: 'Tahci',
//   tittle: 'Developer JR'
// };

// export const TachiApp = ( props ) => {
export const TachiApp = ( { title, subTitle, name } ) => {

  // console.log(props);
   
  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      {/* <h1> { props.title } </h1> */}
      {/* <h1> { title } </h1>
      <h1> { getSuma(21,12)} </h1>
      <code> { JSON.stringify( newValue  ) } </code>
      <p>Soy Developer Jr</p> */}
      <p>{subTitle}</p>
      <p>{subTitle}</p>
      <p>{name}</p>
    </>
  )
}

TachiApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string
}

TachiApp.defaultProps = {
  name: 'Tachi Dito',
  subTitle: 'No hay subtitulo',
  title: 'No hay titulo',
}