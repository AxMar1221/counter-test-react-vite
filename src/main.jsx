import React from 'react';
import ReactDOM  from 'react-dom/client';
// import { CounterApp } from './CounterApp';
import { TachiApp } from './TachiApp';
// import { Helloapp } from './Helloapp';
import './styles.css'


ReactDOM.createRoot( document.getElementById('root')).render(
    <React.StrictMode>
        {/* <CounterApp value={ 21 }/> */}
        <TachiApp title={'Hola soy Tachi'}/>
    </React.StrictMode>
);