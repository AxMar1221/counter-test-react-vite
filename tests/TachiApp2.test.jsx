import { render, screen } from "@testing-library/react"
import { TachiApp } from "../src/TachiApp"

describe('Pruebas en TachiApp', () => {
    const title = 'hola soy Tachi';
    const subTitle = 'Soy un subtitulo';

    test('debe de hacer mach con el snapshot', () => {
        const { container } = render( <TachiApp title={ title }/>)
        expect( container ).toMatchSnapshot();
    });
    
    test('debe de mostrar el mensaje "Hola soy Tachi"', () => {
        render( <TachiApp title={ title }/>)
        expect( screen.getByText(title) ).toBeTruthy();
    });

    test('debe de mostrar el titulo en un h1', () => {
        render( <TachiApp title={ title }/>)
        expect( screen.getByRole('heading',{ level: 1}).innerHTML ).toContain( title );
    });

    test('debe de mostrat el subtitilo enviado por props', () => {
        render( 
            <TachiApp 
                title={ title }
                subTitle={ subTitle}
            />);
        expect( screen.getAllByText( subTitle ).length ).toBe(2);
    });
    
});