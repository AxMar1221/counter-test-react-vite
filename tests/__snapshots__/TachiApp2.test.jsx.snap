// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en TachiApp debe de hacer mach con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    hola soy Tachi
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    No hay subtitulo
  </p>
  <p>
    Tachi Dito
  </p>
</div>
`;
